### PR TITLE
Add csv endpoint wrapper

### LINKED ISSUE #36: Add CSV endpoint wrapper
## Is your feature request related to a problem? Please describe.
Ahead of user testing we've added a simple way to just grab the full file in CSV format, no IDs, no querying, just all the data. We should add this into the package and also our vignette as the absolute easiest / quickest way to grab data out of the API.

## Describe the solution you'd like
Probably it's own separate function, something like `download_full_dataset()`, with API version, data set version (placeholder for now as that isn't something you can query yet), data set ID and verbose as the params.

## Describe alternatives you've considered
Could build in as an option to `query_dataset()` though here you're not really querying. Could also call it `get_dataset()` but could get confused with the GET endpoint more widely.

## Additional context
Here's some starter code to show grabbing from the endpoint:

```
url <- "https://dev.statistics.api.education.gov.uk/api/v1/data-sets/7c0e9201-c7c0-ff73-bee4-304e731ec0e6/csv"
response <- httr::GET(url)
output <- httr::content(response)
```

### PR DESCRIPTION
**Pull request by cjrace**
# Brief overview of changes

Add a wrapper function for getting a CSV version of an API data set from EES.

## Why are these changes being made?

There's a GET endpoint for datasets where you can specify CSV as the response type. There's no querying or even dataset version options on this endpoint yet so it's literally just a download of the latest version.

## Detailed description of changes

Updated `api_url()` to have a new response_type argument where you can switch between JSON or CSV, expecting we can build on this more if the CSV response type gets interest and requests for more features.

I figured `api_url()` made the most sense for these changes, though happy to be told this jars with a different vision you have @rmbielby - aware you've put a lot of thought into the modularising of the code in the package.

## Additional information for reviewers

Couple of small bits of tidy up.

1. Set package to depend on R version 3.5.0 or higher in DESCRIPTION

Did this as there were messages appearing in `devtools::check()` about this related to the test data we have in the package.

Debated adding an on load warning for users like this in a zzz.R file:
```
.onLoad <- function(libname, pkgname) {
  if (getRversion() < "3.5.0") {
    stop("This package requires R version 3.5.0 or later.")
  }
}
```
But decided against as overkill, as it probably won't affect most users.

2. readr as a suggested import

httr uses readr behind the scenes, not including it was causing errors when running tests / examples, but including as a full import was giving a warning as we never call readr directly. This seemed to work as a solution that appeased the R package gods.

3. Added an internal `toggle_message()` function

Now, I know we have this in dfeR, though I wanted to avoid having all of dfeR and it's dependencies and dependencies on this. 

Felt like a small enough function to copy over and stick in a utils.R file for us to use ourselves to control verbosity. Expect we'd take a similar approach with other packages too.

4. upped cyclocomp limits for lintr

We were exceeding on a couple of functions. I think we're mostly safe with what we're doing for now and are modularising a lot already so a higher limit of 25 seemed more suitable and hushed lintr for the time being.

5. I ran `usethis::use_tidy_description()`

It moved some stuff about in the DESCRIPTION file.

## Issue ticket number/s and link

Resolves #36.

### CODE REVIEW

**Review by rmbielby — COMMENTED** [review]
Had a quick scan through and looks good. Was a bit thrown when you hadn't just added a specific "get-csv" endpoint to api_url, which would have been my first thought, but think this way makes sense.

Main thought is handling the additional potential for redundant parameters in `api_url()`. I flag a warning if the time_periods, indicators, etc parameters are set when someone's running an endpoint that doesn't use them, e.g. "get-meta". At the very least I think that flag should trigger for this scenario as well. Plus response_format = CSV doesn't apply to any other endpoint, so should probably be a warning if someone sets it to CSV for any of those requests.

I'm a little wary that if this is the only endpoint that's ever going to be able to deliver a csv, then response_format is a bit redundant and just "get-csv" as an endpoint might make more sense and make the code simpler in terms of having one less thing that doesn't do anything across all the other endpoints.

**Comment by cjrace** [comment]
Interesting to see where your brain instinctively jumped to. I was in two minds when I approached it and this was enough to make me go back and review it a bit more in the API docs.

While in my head this might in the long run just be an option on the `/query` endpoints for response type, currently in the API itself it has been implemented as a separate `/csv` endpoint so for now I've switched over to that, treating it as a `get-csv` endpoint in the code, and we can worry about `response_type` at a later point once we know the future shape of the API better.

Have also made sure the warnings appear for unecessary arguments and added a quick test for it too.

File: R/api_url.R [hunk 1]
```diff
@@ -92,7 +93,8 @@ api_url <- function(
       stop(
         paste(
           "You have entered an invalid endpoint, this should one of:",
-          "get-summary, get-meta, get-data or post-data"
+          "get-publication, get-data-catalogue, get-summary, get-meta,",
```
Comment (rmbielby) [inline R/api_url.R:96]: Slight typo here, should be "get-publications" rather then "get-publication"

**Review by rmbielby — CHANGES_REQUESTED** [review]

File: R/api_url.R (hunk 1 above)
Comment (cjrace) [reply R/api_url.R]: Good spot! Fixed that now

**Review by rmbielby — APPROVED** [review]

**Pull request merged**